magento/magento2
# 38325 make city field optional in address

### Description (*)
In this PR added the field to the configuration to make the field 'City' not required in the forms. Also changed the shipping address form, and customer register form, and added the MFTF test for the field in configuration.

### Related Pull Requests
There is no related PR

### Fixed Issues (if relevant)
1. Fixes magento/magento2#38325

### Manual testing scenarios (*)
1. Change the configuration for the City field in Configuration -> Customers -> Name and Address Options -> Show City
2. Check on the frontend on the checkout shipping address step, customer address edit form.

### Questions or comments

### Contribution checklist (*)
